lib/filewriter.js

```diff
@@ -138,16 +138,18 @@ FileWriter.prototype = {
             var guard, error = null;
             function finish(err) {
                 if (guard && global.clearTimeout) clearTimeout(guard);
+                var err2;
                 guard = null;
-                if (fd !== null) fs.close(fd, function(){});
+                if (fd !== null) try { fs.closeSync(fd) } catch (e) { err2 = e }
```

Comment:
What about naming this more clearly, something like `fileCloseError` instead of `err2`? same comment for the changed code below.